kpops/component_handlers/kafka_connect/connect_wrapper.py

```diff
@@ -63,17 +63,18 @@ def create_connector(
             self.create_connector(connector_config)
         raise KafkaConnectError(response)
 
-    def get_connector(self, connector_name: str) -> KafkaConnectResponse:
+    def get_connector(self, connector_name: str | None) -> KafkaConnectResponse:
         """Get information about the connector.
 
-        API Reference:
-        https://docs.confluent.io/platform/current/connect/references/restapi.html#get--connectors-(string-name)
-
+        API Reference: https://docs.confluent.io/platform/current/connect/references/restapi.html#get--connectors-(string-name)
         :param connector_name: Nameof the crated connector
         :return: Information about the connector.
         """
+        if connector_name is None:
+            msg = "Connector name not set"
+            raise Exception(msg)
```

Comment:
Because I changed the function signature. It also accepts `None` for `connector_name` now. Otherwise `pyright` complains about the following:

<details>

<summary>Error message</summary>

```

kpops/kpops/component_handlers/kafka_connect/kafka_connect_handler.py
  /home/sujuka99/Projects/bakdata/kpops/kpops/component_handlers/kafka_connect/kafka_connect_handler.py:51:65 - error: Argument of type "str | None" cannot be assigned to parameter "connector_name" of type "str" in function "get_connector"
    Type "str | None" cannot be assigned to type "str"
      Type "None" cannot be assigned to type "str" (reportGeneralTypeIssues)
kpops/kpops/component_handlers/kafka_connect/kafka_connect_handler.py:97:61 - error: Argument of type "str | None" cannot be assigned to parameter "connector_name" of type "str" in function "get_connector"
    Type "str | None" cannot be assigned to type "str"
      Type "None" cannot be assigned to type "str" (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 informations

```

</details>

Had to do it because I changed the `name` attr [here](https://github.com/bakdata/kpops/commit/b4f6aa0efdffcd6cbdfeef23143c3b9aaff189f6#diff-2db540fa7eaed4196f822009ae6ba5ee0e057926beba7ce89d827990e9e65eafR19-R20)


Changed it there, because after some "migrating" I encountered the following error:

<details>

<summary>Error message</summary>

```

connector_config = KafkaConnectorConfig(connector_class='com.bakdata.connect.TestConnector', name='${pipeline_name}-test-connector-with-long-name-0123456789abcdefghijklmnop')        [10/1871]

    def test_connector_config_name_override(
        self,
        config: KpopsConfig,
        handlers: ComponentHandlers,
        connector_config: KafkaConnectorConfig,
    ):
        connector = KafkaConnector(
            name=CONNECTOR_NAME,
            config=config,
            handlers=handlers,
            app=connector_config,
            namespace="test-namespace",
        )
        assert connector.app.name == CONNECTOR_FULL_NAME

>       connector = KafkaConnector(
            name=CONNECTOR_NAME,
            config=config,
            handlers=handlers,
            app={"connector.class": CONNECTOR_CLASS},  # type: ignore[reportGeneralTypeIssues]
            namespace="test-namespace",
        )

tests/components/test_kafka_connector.py:79:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
kpops/components/base_components/pipeline_component.py:58: in __init__
    super().__init__(**kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = KafkaConnector()
kwargs = {'app': {'connector.class': 'com.bakdata.connect.TestConnector'}, 'config': KpopsConfig(defaults_path=PosixPath('/home...s.ComponentHandlers object at 0x7fa75c97bd50>, 'name': 'test-
connector-with-long-name-0123456789abcdefghijklmnop', ...}

    def __init__(self, **kwargs) -> None:
        if kwargs.get("enrich", True):
            kwargs = self.extend_with_defaults(**kwargs)
>       super().__init__(**kwargs)
E       pydantic_core._pydantic_core.ValidationError: 1 validation error for KafkaConnector
E       app.name
E         Field required [type=missing, input_value={'connector.class': 'com.....connect.TestConnector'}, input_type=dict]
E           For further information visit https://errors.pydantic.dev/2.4/v/missing

kpops/components/base_components/base_defaults_component.py:74: ValidationError

```

</details>

Reply:
Maybe this is linked to the name-trimming problem? I don't remember anymore why I handled the initial problem like that...